src/NavigationMock.tsx

```diff
@@ -47,35 +54,34 @@ class NativeNavigationMock {
   }
 
   push(componentId: string, { component }: { component: LayoutComponent }) {
-    // call the current screen disappear event
-    this.callComponentDidDisappear(componentId);
+    this.componentIdCounter += 1;
 
-    // push the new screen
-    this.screenStack.push(component);
+    const currentScreenComponentId = this.currentScreen?.componentId;
+    if (currentScreenComponentId) this.callComponentDidDisappear(currentScreenComponentId);
```

Comment:
Best practice in my opinion is to always wrap if with curly braces `{}`. 
One exception to this rule is if you return immediately e.g
```
if (ture) return;
```